From 4ad05133a9d5ab9af124dc008167719bc7d39ffe Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Sat, 10 Feb 2018 22:14:57 +0100
Subject: [PATCH] do not escape double quotes

they would end up escaped in the pluginoutput and break html output for example.

Signed-off-by: Sven Nierlein <sven@nierlein.de>
---
 src/naemon/checks.c | 2 +-
 tests/test-checks.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/naemon/checks.c b/src/naemon/checks.c
index f86299a9..73e5e459 100644
--- a/src/naemon/checks.c
+++ b/src/naemon/checks.c
@@ -188,7 +188,7 @@ int parse_check_output(char *buf, char **short_output, char **long_output, char
 	*short_output = check_output->short_output;
 	*perf_data = check_output->perf_data;
 	if(escape_newlines_please == TRUE && check_output->long_output != NULL) {
-		*long_output = g_strescape(check_output->long_output, "");
+		*long_output = g_strescape(check_output->long_output, "\"");
 		free(check_output->long_output);
 	} else {
 		*long_output = check_output->long_output;
diff --git a/tests/test-checks.c b/tests/test-checks.c
index 4409680e..ac83ae99 100644
--- a/tests/test-checks.c
+++ b/tests/test-checks.c
@@ -168,11 +168,11 @@ START_TEST(multiple_line_output_newline_escaping)
 {
 	full_output = "TEST OK - ...\n"
 				  "Here's a second line of output and\n"
-				  "one more\n";
+				  "one \"more\"\n";
 	output = strdup(full_output);
 	parse_check_output(output, &short_output, &long_output, &perf_data, TRUE, FALSE);
 	ck_assert_str_eq("TEST OK - ...", short_output);
-	ck_assert_str_eq("Here's a second line of output and\\none more\\n", long_output);
+	ck_assert_str_eq("Here's a second line of output and\\none \"more\"\\n", long_output);
 }
 END_TEST
 
